src/FakeItEasy/Configuration/AnyCallCallRule.cs

```diff
@@ -39,9 +39,9 @@ protected override bool OnIsApplicableTo(IFakeObjectCall fakeObjectCall)
         {
             return
                 this.argumentsPredicate(fakeObjectCall.Arguments) &&
-                    (this.ApplicableToMembersWithReturnType == null
-                    || (this.ApplicableToMembersWithReturnType == fakeObjectCall.Method.ReturnType)
-                    || (this.ApplicableToAllNonVoidReturnTypes && fakeObjectCall.Method.ReturnType != typeof(void)));
+                ((this.ApplicableToMembersWithReturnType == null && !this.ApplicableToAllNonVoidReturnTypes)
+                 || (this.ApplicableToMembersWithReturnType == fakeObjectCall.Method.ReturnType)
+                 || (this.ApplicableToAllNonVoidReturnTypes && fakeObjectCall.Method.ReturnType != typeof(void)));

```

Comment:
This should work, but I'm worried this statement is becoming hard to understand. Maybe it would be better to change it to something like this:
```
if (this.argumentsPredicate(fakeObjectCall.Arguments))
{
    if (ApplicableToAllNonVoidReturnTypes)
    {
        return fakeObjectCall.Method.ReturnType != typeof(void);
    }
    else if (ApplicableToMembersWithReturnType != null)
    {
        return this.ApplicableToMembersWithReturnType == fakeObjectCall.Method.ReturnType;
    }
    else
    {
        return true;
    }
}
return false;
```